### PR TITLE
Add basic web device and run support

### LINKED ISSUE #28287: Support for minimal static asset serving from the build/ directory
Required for web based builds, could be a simple shelf server for now.

### LINKED ISSUE #28288: Add web device and support `run --web`.
Should be disabled behind a flag. Run should attempt a dartdevc build and fallback to dart2js.

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Adds a skeleton web device, web specific project, and support for run via `--release --no-hot`. This device exists behind the FLUTTER_WEB env variable to prevent breaking default flutter run until we decide on the UX there. Added ability to launch chrome on mac.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/28288
Fixes https://github.com/flutter/flutter/issues/28287
Part of https://github.com/flutter/flutter/issues/28285

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes tests for *all* changed/updated/fixed behaviors (See [Test Coverage]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
For windows, I believe the command is as simple as `start chrome.exe` - but this is really just a placeholder for now.

**Comment by jonahwilliams** [comment]
friendly ping @yjbanov

**Review by yjbanov — APPROVED** [review]
lgtm

File: packages/flutter_tools/lib/src/project.dart [hunk 1]
```diff
@@ -454,6 +461,31 @@ class AndroidProject {
...
 
+/// Represents the web sub-project of a Flutter project.
+class WebProject {
+  WebProject._(this.parent);
+
+  final FlutterProject parent;
+
+   Future<void> ensureReadyForPlatformSpecificTooling() async {
+    /// Generate index.html in build/web. Eventually we could support
+    /// a custom html under the web sub directory.
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/project.dart]: Makes sense, we can generate the index.html only as a part of flutter create and then separate out the generatedPluginRegistry files.

**Pull request merged**